ports/esp32/main.c

```diff
@@ -219,6 +219,12 @@ void mp_task(void *pvParameter) {
 
     mp_deinit();
     fflush(stdout);
+
+    #if MICROPY_HW_ENABLE_USB_RUNTIME_DEVICE
+    // matches mp_usbd_deinit() above, ensure we have USB-CDC after soft reset
+    mp_usbd_init();
```

Comment:
Do you think it makes sense to move this to the top of the soft reset loop, after `mp_init()`?  And then remove the call to `mp_usbd_init()` from `usb.c:usb_init()`?

That would then match most of the other ports (eg rp2) in the way they call `mp_usbd_init` and `mp_usbd_deinit`.

Reply:
Done! I also realised another problem with the init order.